src/features/autosuggest/autosuggest.ts

```diff
@@ -15,14 +25,11 @@ export async function autoSuggest(
   options: AutosuggestOptions,
 ): Promise<SuggestResponse> {
   const { suggestEndpoint, ...config } = configuration;
-  const fixed: SuggestFixedOptions = {
-    request_type: 'suggest',
-  };
   const defaults: Partial<SuggestRequestParameters> = {};
-  const queryParams = Object.assign(config, fixed, defaults, options);
+  const queryParams = Object.assign(config, FIXED_OPTIONS, defaults, options);
```

Comment:
Should the `FIXED_OPTIONS` be at the end, so they can't be overridden by the user? What is the goal of separating the `FIXED_OPTIONS` from the `defaults`?

Reply:
Yes I think thats an improvement! 

Wrt the `FIXED_OPTIONS` vs `defaults`: I separated them so that we can create that runtime check for the options object. The `FIXED_OPTIONS` are truly unique in identifying which API will be used in the backend, the default options are not.